src/index.ts

```diff
@@ -7,20 +7,39 @@ import { Eslint, Prettier, Rails } from './commands'
 program.version('0.0.dev')
 
 program
-  .command('prettier')
-  .description('Install Prettier with Gnarly Opinions')
-  .action(Prettier.run)
+  .command('add')
+  .argument('<packageName>', 'the package name')
+  .description('Add and Setup Dependencies with Gnarly Opinions')
+  .action(packageName => {
+    switch (packageName.toLowerCase()) {
+      case 'prettier':
+        Prettier.add()
+        break
+      case 'eslint':
+        Eslint.add()
+        break
+      default:
+        process.stdout.write(`Unrecognized package name passed to gnar add: ${packageName}`)
+        break
+    }
+  })
 
 program
-  .command('eslint')
-  .description('Install ESlint + Prettier with Gnarly Opinions')
-  .action(Eslint.run)
+  .command('init')
+  .argument('<frameworkName>', 'the framework name')
+  .argument('<subCommand>', 'subcommand')
+  .description('Greenfield Frameworks with Gnarly Opinions')
+  .action((frameworkName, subCommand, _options, command) => {
+    const extraArgs = command.args.filter((arg: string) => arg !== subCommand).join(' ')
 
-program
-  .command('rails')
-  .argument('<mainCommand>', 'first argument passed to rails')
-  .argument('[subcommand]', 'Second command passed to rails')
-  .description('install ESLint with Gnarly configuration - now with Prettier!')
-  .action(Rails.run)
+    switch (frameworkName.toLowerCase()) {
+      case 'rails':
+        Rails.init(subCommand, extraArgs)
+        break
+      default:
+        process.stdout.write(`Unrecognized framework name passed to gnar add: ${frameworkName}`)
```

Comment:
Should we list the permissible framework names?

Reply:
Listed!